x/rewards/client/cli/tx.go

```diff
@@ -32,3 +45,106 @@ func GetTxCmd() *cobra.Command {
 
 	return cmd
 }
+
+// SetIprpcDataProposalHandler is the param change proposal handler.
+var SetIprpcDataProposalHandler = govclient.NewProposalHandler(NewSubmitSetIprpcDataProposalTxCmd)
+
+// NewSubmitSetIprpcDataProposalTxCmd returns a CLI command handler for creating
+// a set-iprpc-data proposal governance transaction.
+func NewSubmitSetIprpcDataProposalTxCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "set-iprpc-data <deposit> --min-cost 0ulava --add-subscriptions addr1,addr2 --remove-subscriptions addr3,addr4",
+		Args:  cobra.ExactArgs(1),
+		Short: "Submit a set IPRPC data proposal",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Submit a set IPRPC data proposal along with an initial deposit.
+The proposal details (IPRPC data to set) are supplied via optional flags. To set the minimum IPRPC cost, use the --min-cost flag.
+To add IPRPC eligible subscriptions use the --add-subscriptions flag. To remove IPRPC eligible subscriptions use the --remove-subscriptions flag.
+Optionally, you can make the proposal expedited using the --expedited flag.
+Example:
+$ %s tx gov submit-legacy-proposal set-iprpc-data --min-cost 0ulava --add-subscriptions addr1,addr2 --remove-subscriptions addr3,addr4
+`,
+				version.AppName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			from := clientCtx.GetFromAddress()
+
+			isExpedited, err := cmd.Flags().GetBool(expeditedFlagName)
+			if err != nil {
+				return err
+			}
+
+			// get min cost
+			costStr, err := cmd.Flags().GetString(minIprpcCostFlagName)
```

Comment:
either make this flag mendatory or make sure if the flag isnt used to get the current mincost and set it in the message

Reply:
done. see https://github.com/lavanet/lava/commit/3db52d234bc3570cf3139ef15b60a683ac9ffd88